kv/store.go

```diff
@@ -0,0 +1,71 @@
+package kv
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"go.uber.org/zap"
+
+	"github.com/segmentio/ksuid"
+	"github.com/serverless/event-gateway-connector/connection"
+	"go.etcd.io/etcd/clientv3"
+)
+
+// Store implements connection.Service using etcd KV as a backend.
+type Store struct {
+	Client clientv3.KV
+	Log    *zap.SugaredLogger
+}
+
+var _ connection.Service = (*Store)(nil)
+
+// CreateConnection creates connection in etcd.
+func (store Store) CreateConnection(conn *connection.Connection) (*connection.Connection, error) {
+	id, err := ksuid.NewRandom()
+	if err != nil {
+		return nil, err
+	}
+	conn.ID = connection.ID(id.String())
+
+	byt, err := json.Marshal(conn)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = store.Client.Put(context.TODO(), connectionKey{Space: conn.Space, ID: conn.ID}.String(), string(byt))
```

Comment:
Does `etcd` require stringify of payload? Can you not simply store the byte stream?

Reply:
etcd accepts strings as a value. it's weird, I know.